site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -2365,6 +2365,78 @@ public function saveGradedComponent(TaGradedGradeable $ta_graded_gradeable, Grad
         }
     }
 
+    /**
+     * Route for saving the marks the submitter received for a component
+     */
+    #[Route("/courses/{_semester}/{_course}/gradeable/{gradeable_id}/grading/graded_gradeable/change_grade_version", methods: ["POST"])]
+    public function ajaxUpdateGradedVersionForStudent(string $gradeable_id): ?Jsonresponse {
+        $anon_id = $_POST['anon_id'] ?? null;
+        $graded_version = intval($_POST['graded_version'] ?? null);
+        $component_ids = $_POST['component_ids'] ?? [];
+
+        if ($anon_id === null) {
+            return JsonResponse::getFailResponse('Missing anon_id parameter');
+        }
+        if ($graded_version < 1) {
+            return JsonResponse::getFailResponse('Invalid graded_version parameter');
+        }
+
+        if (count($component_ids) < 1) {
+            return JsonResponse::getFailResponse('Missing component_ids parameter');
+        }
+
+        // Get the gradeable
+        $gradeable = $this->tryGetGradeable($gradeable_id);
+        if ($gradeable === false) {
+            return null;
+        }
+        // Get user id from the anon id
+        $submitter_id = $this->tryGetSubmitterIdFromAnonId($anon_id, $gradeable_id);
+        if ($submitter_id === false) {
+            return null;
+        }
+
+        // Get the graded gradeable
+        $graded_gradeable = $this->tryGetGradedGradeable($gradeable, $submitter_id);
+        if ($graded_gradeable === false) {
+            return null;
+        }
+
+
+        // check if all the components provided are real and the user has permission
+        foreach ($component_ids as &$component_id) {
```

Comment:
```suggestion
        foreach ($component_ids as $component_id) {
```
Should we remove `&` from foreach ($component_ids as &$component_id) since there was no modification of `$component_id` in the loop?